plugins/com.aptana.editor.common/src/com/aptana/editor/common/preferences/EditorsPreferencePage.java

```diff
@@ -81,6 +81,10 @@ public void createFieldEditors()
 		addField(new BooleanFieldEditor(IPreferenceConstants.ENABLE_WORD_WRAP,
 				Messages.EditorsPreferencePage_Enable_WordWrap, appearanceComposite));
 
+		//Open no extension files with JavaScript source editor.
+		addField(new BooleanFieldEditor(IPreferenceConstants.OPEN_WITH_JS_EDITOR,
+						Messages.EditorsPreferencePage_OpenWith_JSEditor, appearanceComposite));
+
```

Comment:
I would prefer to add a more generic preference option. We can query the existing editor contributions and provide a drop down with the list of the editor names. User can chose any of the editor for the alien files.